anonymizer.go

```diff
@@ -0,0 +1,40 @@
+package goanonymizer
+
+import (
+	"errors"
+	"net"
+)
+
+//Anonymize struct that hold mask for V4 and V6.
+type Anonymize struct {
+	v4Mask net.IPMask
+	v6Mask net.IPMask
+}
+
+//NewAnonymize returns instance of Anonymize.
+//If v4Mask and v6Mask are not provided, default values are used.
+func NewAnonymize(v4Mask, v6Mask string) *Anonymize {
+	ipV4 := net.ParseIP(v4Mask)
+	if ipV4 == nil {
+		ipV4 = net.ParseIP("255.255.255.0")
+	}
+	ipV6 := net.ParseIP(v6Mask)
+	if ipV6 == nil {
+		ipV6 = net.ParseIP("ffff:ffff:ffff:ffff:0000:0000:0000:0000")
+	}
+
+	return &Anonymize{v4Mask: net.IPMask(ipV4.To4()), v6Mask: net.IPMask(ipV6)}
+}
+
+//AnonymizeIp gets IP and transform it based on v4Mask and v6Mask.
+func (a Anonymize) AnonymizeIp(ipStr string) (string, error) {
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		return "", errors.New("IP is not valid")
+	}
+	if ipv4 := ip.To4(); ipv4 != nil {
```

Comment:
I think a type switch reads a lot better:
```go
switch v := ip.(type) {
    case ipv4:
        return ...
    case ipv6:
        return ...
}
```

Reply:
does not work, it's the same type :-|